vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -552,28 +553,35 @@ void AdsImpl::SetConfirmationsIsReady(
 }
 
 std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory(
-    const AdsHistoryFilterType ads_history_filter_type) {
-  std::map<uint64_t, std::vector<AdsHistory>> ads_history;
-  base::Time now = base::Time::Now().LocalMidnight();
+    const AdsHistoryFilterType filter_type,
+    const AdsHistorySortType sort_type) {
+  auto history = client_->GetAdsShownHistory();
 
-  auto ad_history_details = client_->GetAdsShownHistory();
+  auto sort = AdsHistorySortFactory::Build(sort_type);
+  if (sort) {
+    history = sort->Apply(history);
+  }
```

Comment:
It would be more efficient to apply the sort after the filter.

Reply:
The sort is after the filter, however, I have made the code more readable by refactoring the filter to use a factory (as previously this was out of the scope of past work)